evalml/pipelines/components/estimators/classifiers/lightgbm_classifier.py

```diff
@@ -189,8 +189,11 @@ def fit(self, X, y=None):
             self
         """
         X = infer_feature_types(X)
-        X_encoded = self._encode_categories(X, fit=True)
-        y_encoded = self._encode_labels(y)
+        if y is not None:
+            y = infer_feature_types(y)
+        X_d, y_d = self._handle_nullable_types(X, y)
```

Comment:
Ah, I kept meaning to bring the `_d` thing up - it's a way of indicating the data has been downcasted that I wanted to introduce (like `_t` indicates "has been transformed" and `_ww` "has been made into woodwork". If people dislike it, I can remove it.

In general, I wasn't particularly following any way of choosing to overwrite variable names or not. I was more following what the component itself was already doing. So the lgbm estimators weren't overwriting variable names (like the lgbm estimators chose a different `X_encoded` instead of overwriting `X` while exponental smoothing was doing stuff like `y = self._remove_datetime(y)`, so I kind of just tried to follow patterns. 

We seem to have a good mixture of how we decide to overwrite variable names across our components, so I didn't necessarily feel like this was the time to exert my hand at standardizing that. But also given that I'm attempting to introduce `_d`, I can see how keeping things standard would be extra important. Happy to standardize things if you feel like it'd clarify things up!

Reply:
I see what you mean. Seems like standardizing if/when we copy vs overwrite data within transformers is a great future tech debt item 🥲 

I don't have a strong opinion one way or the other as to whether we standardize the use of `_d` here. I do wonder about downstream impacts though - would this be related to the odd behavior in the docs at all?